src/ErrorBoundary.js

```diff
@@ -66,4 +68,16 @@ class ErrorBoundary extends Component {
   }
 }
 
+
+export const withErrorBoundary: Function = (
```

Comment:
Remove the `: Function` declaration. This prevents Flow from actually evaluating with the more strict type (that it can infer from the right hand side of the statement).